### DISCUSSION TITLE
WPC MET MODE Errors

### DISCUSSION THREAD

**Discussion by m-rgreen** [dtcenter/METplus#2362] [Installation]
Errors when trying to run MET v11.1 MODE:

/opt/MET/METPlus/bin/mode: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0)
/opt/MET/METPlus/bin/mode: /lib64/libc.so.6: version `GLIBC_2.25' not found (required by /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0)
/opt/MET/METPlus/bin/mode: /lib64/libc.so.6: version `GLIBC_2.26' not found (required by /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0)
/opt/MET/METPlus/bin/mode: /lib64/libc.so.6: version `GLIBC_2.27' not found (required by /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0)
/opt/MET/METPlus/bin/mode: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by /opt/MET/METPlus/external_libs/lib/libhdf5.so.200)

Output from "ldd /opt/MET/METPlus/bin/mode":

/opt/MET/METPlus/bin/mode: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.8' not found (required by /opt/MET/METPlus/bin/mode)
/opt/MET/METPlus/bin/mode: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.20' not found (required by /opt/MET/METPlus/bin/mode)
/opt/MET/METPlus/bin/mode: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /opt/MET/METPlus/bin/mode)
/opt/MET/METPlus/bin/mode: /usr/lib64/libc.so.6: version `GLIBC_2.28' not found (required by /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0)
/opt/MET/METPlus/bin/mode: /usr/lib64/libc.so.6: version `GLIBC_2.25' not found (required by /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0)
/opt/MET/METPlus/bin/mode: /usr/lib64/libc.so.6: version `GLIBC_2.26' not found (required by /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0)
/opt/MET/METPlus/bin/mode: /usr/lib64/libc.so.6: version `GLIBC_2.27' not found (required by /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0)
/opt/MET/METPlus/bin/mode: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /opt/MET/METPlus/external_libs/lib/libnetcdf_c++4.so.1)
/opt/MET/METPlus/bin/mode: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.21' not found (required by /opt/MET/METPlus/external_libs/lib/libnetcdf_c++4.so.1)
/opt/MET/METPlus/bin/mode: /usr/lib64/libm.so.6: version `GLIBC_2.27' not found (required by /opt/MET/METPlus/external_libs/lib/libhdf5.so.200)
        linux-vdso.so.1 =>  (0x00007ffc99bc9000)
        libpython3.10.so.1.0 => /opt/MET/external/Python-3.10.4/lib/libpython3.10.so.1.0 (0x00007f0011fda000)
        libcrypt.so.1 => /usr/lib64/libcrypt.so.1 (0x00007f0011da3000)
        libpthread.so.0 => /usr/lib64/libpthread.so.0 (0x00007f0011b87000)
        libdl.so.2 => /usr/lib64/libdl.so.2 (0x00007f0011983000)
        libutil.so.1 => /usr/lib64/libutil.so.1 (0x00007f0011780000)
        libm.so.6 => /usr/lib64/libm.so.6 (0x00007f001147e000)
        libpng16.so.16 => /opt/MET/METPlus/external_libs/lib/libpng16.so.16 (0x00007f0012783000)
        libz.so.1 => /opt/MET/METPlus/external_libs/lib/libz.so.1 (0x00007f0012765000)
        libnetcdf_c++4.so.1 => /opt/MET/METPlus/external_libs/lib/libnetcdf_c++4.so.1 (0x00007f0012715000)
        libnetcdf.so.18 => /opt/MET/METPlus/external_libs/lib/libnetcdf.so.18 (0x00007f00112fa000)
        libgsl.so.19 => /opt/MET/METPlus/external_libs/lib/libgsl.so.19 (0x00007f00110cc000)
        libgslcblas.so.0 => /opt/MET/METPlus/external_libs/lib/libgslcblas.so.0 (0x00007f00126d2000)
        libhdf5_hl.so.200 => /opt/MET/METPlus/external_libs/lib/libhdf5_hl.so.200 (0x00007f00126ad000)
        libhdf5.so.200 => /opt/MET/METPlus/external_libs/lib/libhdf5.so.200 (0x00007f0010ca3000)
        libstdc++.so.6 => /usr/lib64/libstdc++.so.6 (0x00007f001099b000)
        libgomp.so.1 => /usr/lib64/libgomp.so.1 (0x00007f0010775000)
        libgcc_s.so.1 => /usr/lib64/libgcc_s.so.1 (0x00007f001055f000)
        libc.so.6 => /usr/lib64/libc.so.6 (0x00007f0010191000)
        libfreebl3.so => /usr/lib64/libfreebl3.so (0x00007f000ff8e000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f001259a000)
        libcurl.so.4 => /usr/lib64/libcurl.so.4 (0x00007f000fd24000)
        libidn.so.11 => /usr/lib64/libidn.so.11 (0x00007f000faf1000)
        libssh2.so.1 => /usr/lib64/libssh2.so.1 (0x00007f000f8c4000)
        libssl3.so => /usr/lib64/libssl3.so (0x00007f000f65f000)
        libsmime3.so => /usr/lib64/libsmime3.so (0x00007f000f437000)
        libnss3.so => /usr/lib64/libnss3.so (0x00007f000f0fd000)
        libnssutil3.so => /usr/lib64/libnssutil3.so (0x00007f000eecd000)
        libplds4.so => /usr/lib64/libplds4.so (0x00007f000ecc9000)
        libplc4.so => /usr/lib64/libplc4.so (0x00007f000eac4000)
        libnspr4.so => /usr/lib64/libnspr4.so (0x00007f000e885000)
        libgssapi_krb5.so.2 => /usr/lib64/libgssapi_krb5.so.2 (0x00007f000e638000)
        libkrb5.so.3 => /usr/lib64/libkrb5.so.3 (0x00007f000e34f000)
        libk5crypto.so.3 => /usr/lib64/libk5crypto.so.3 (0x00007f000e11c000)
        libcom_err.so.2 => /usr/lib64/libcom_err.so.2 (0x00007f000df18000)
        liblber-2.4.so.2 => /usr/lib64/liblber-2.4.so.2 (0x00007f000dd09000)
        libldap-2.4.so.2 => /usr/lib64/libldap-2.4.so.2 (0x00007f000dab4000)
        libssl.so.10 => /usr/lib64/libssl.so.10 (0x00007f000d842000)
        libcrypto.so.10 => /usr/lib64/libcrypto.so.10 (0x00007f000d3df000)
        librt.so.1 => /usr/lib64/librt.so.1 (0x00007f000d1d7000)
        libkrb5support.so.0 => /usr/lib64/libkrb5support.so.0 (0x00007f000cfc7000)
        libkeyutils.so.1 => /usr/lib64/libkeyutils.so.1 (0x00007f000cdc3000)
        libresolv.so.2 => /usr/lib64/libresolv.so.2 (0x00007f000cba9000)
        libsasl2.so.3 => /usr/lib64/libsasl2.so.3 (0x00007f000c98c000)
        libselinux.so.1 => /usr/lib64/libselinux.so.1 (0x00007f000c765000)
        libpcre.so.1 => /usr/lib64/libpcre.so.1 (0x00007f000c503000)

**Comment by jprestop** [comment]
Thank you @m-rgreen and @anroy12 for passing along this information.  I am sorry to hear about the trouble.  Could you please confirm which python executable they are using?  It would also be good to check what operating system they are using. Could you please have them run the following and send the output? 

```
/etc/os-release
```

Thank you!

**Comment by m-rgreen** [comment]
We are do not have permissions to test that command line. We are also doing some tests to test a theory that the different OS versions on the machines may be contributing to some of the errors.

**Reply by jprestop** [reply]
@m-rgreen Ugh apologies - the command should be `more /etc/os-release`. You should have permissions for that, I believe.... And, we do believe that the different OS versions on the machines may be contributing to some of the errors.

**Reply by m-rgreen** [reply]
The output here is what we are thinking about. One of the new operational VMs that has not given the initial errors is on RHEL 8, and our local linux workstations are on RHEL 7. 

Local linux workstation output:
NAME="Red Hat Enterprise Linux Workstation"
VERSION="7.9 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Workstation"
VARIANT_ID="workstation"
VERSION_ID="7.9"
PRETTY_NAME="Red Hat Enterprise Linux"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.9:GA:workstation"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.9
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.9"


New VMs output:
NAME="Red Hat Enterprise Linux"
VERSION="8.8 (Ootpa)"
ID="rhel"
ID_LIKE="fedora"
VERSION_ID="8.8"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Red Hat Enterprise Linux 8.8 (Ootpa)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:8::baseos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://access.redhat.com/documentation/en-us/red_hat_enterpr
ise_linux/8"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 8"
REDHAT_BUGZILLA_PRODUCT_VERSION=8.8
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.8"

**Reply by jprestop** [reply]
Thank you @m-rgreen.  Please keep us posted.  I suspect that is likely the problem.

**Reply by jprestop** [reply]
Hi @m-rgreen.  I just wanted to follow up and see if you were able to figure anything out?

**Reply by jprestop** [reply]
Hi @m-rgreen. I just wanted to check in and see if you have been able to get around the problems you were experiencing?

**Reply by m-rgreen** [reply]
Apologies for the delayed response. I will chat with a fellow developer this coming week on how to best test a workaround. I will follow up then.

**Reply by jprestop** [reply]
@m-rgreen It's ok.  I just wanted to make sure this isn't forgotten about, so I thought I'd follow up.  Sounds good!  Thanks!

**Reply by jprestop** [reply]
HI @m-rgreen.  It's been a while since I checked in.  I thought I'd follow up to see if you were able to get around the problems you were experiencing?
